src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,173 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+
+/**
+ * Storage links the TaskList from Duke to the user's local storage.
+ */
+public class Storage {
+    private final Path pathToStorage;
+    private List<String> allTasks;
+
+    /**
+     * Constructor to create a Storage object, which will be used when Duke creates, updates, or deletes Tasks.
+     *
+     * @param pathToStorage the path to the user's storage where the Tasks will be stored.
+     */
+    public Storage(Path pathToStorage) {
+        this.pathToStorage = pathToStorage;
+        assert pathToStorage != null : "Path to storage cannot be null";
+        try {
+            // Create directory if needed
+            Path parentPath = pathToStorage.getParent();
+            Files.createDirectories(parentPath);
+
+            if (!Files.exists(pathToStorage)) {
+                Files.createFile(pathToStorage);
```

Comment:
Perhaps can let the user know that a File is created? So that the user is informed of what other files the app is adding to their laptop.